test/DurableTask.SqlServer.AzureFunctions.Tests/CoreScenarios.cs

```diff
@@ -154,7 +154,155 @@ public async Task MultiInstanceQueries()
+
+            statusFilters.Remove(OrchestrationRuntimeStatus.Running);
+            result = await client.ListInstancesAsync(condition, CancellationToken.None);
+            Assert.Equal(5, result.DurableOrchestrationState.Count());
+            Assert.All(result.DurableOrchestrationState, state => Assert.Equal(OrchestrationRuntimeStatus.Completed, state.RuntimeStatus));
+
+            statusFilters.Clear();
+            result = await client.ListInstancesAsync(condition, CancellationToken.None);
+            Assert.Equal(10, result.DurableOrchestrationState.Count());
+
+            // Test InstanceIdPrefix
+            condition.InstanceIdPrefix = "Foo";
+            result = await client.ListInstancesAsync(condition, CancellationToken.None);
+            Assert.Empty(result.DurableOrchestrationState);
+
+            condition.InstanceIdPrefix = prefix;
+            result = await client.ListInstancesAsync(condition, CancellationToken.None);
+            Assert.Equal(10, result.DurableOrchestrationState.Count());
+
+            // Test PageSize and ContinuationToken
+            var instanceIds = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            condition.PageSize = 0;
+            while (condition.PageSize++ < 10)
+            {
+                result = await client.ListInstancesAsync(condition, CancellationToken.None);
+                int total = result.DurableOrchestrationState.Count();
+                Assert.Equal(condition.PageSize, total);
+
+                // Make sure all instance IDs are unique
+                Assert.All(result.DurableOrchestrationState, s => instanceIds.Add(s.InstanceId));
+
+                while (total < 10)
+                {
+                    condition.ContinuationToken = result.ContinuationToken;
+                    result = await client.ListInstancesAsync(condition, CancellationToken.None);
+                    int count = result.DurableOrchestrationState.Count();
+                    Assert.NotEqual(0, count);
+                    Assert.True(count <= condition.PageSize);
+                    total += count;
+                    Assert.True(total <= 10);
+                    Assert.All(result.DurableOrchestrationState, s => instanceIds.Add(s.InstanceId));
+                }
+
+                condition.ContinuationToken = null;
+            }
+
+            // Test ShowInput
+            condition.ShowInput = true;
+            condition.CreatedTimeFrom = sequencesFinishedTime;
+            result = await client.ListInstancesAsync(condition, CancellationToken.None);
+            Assert.All(result.DurableOrchestrationState, state => Assert.NotEqual(JValue.CreateNull(), state.Input));
+
+            condition.ShowInput = false;
+            condition.CreatedTimeFrom = startTime;
+            result = await client.ListInstancesAsync(condition, CancellationToken.None);
+            Assert.All(result.DurableOrchestrationState, state => Assert.Equal(JValue.CreateNull(), state.Input));
```

Comment:
Based on how the diff worked out, it looks like the core test logic is the same for both your new test and this older test. Is it possible to refactor this common code into a separate static method?

Reply:
I called the original test from the new one as it's more a regression check